lib/client/keystore.go

```diff
@@ -12,255 +12,261 @@ distributed under the License is distributed on an "AS IS" BASIS,
 WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
-
-
-Keystore implements functions for saving and loading from hard disc
-temporary teleport certificates
 */
 
 package client
 
 import (
-	"encoding/json"
+	"fmt"
+	"io"
 	"io/ioutil"
-	"math/rand"
-	"net"
 	"os"
 	"os/user"
 	"path/filepath"
-	"strconv"
-	"strings"
 	"time"
 
-	"github.com/gravitational/teleport/lib/backend/boltbk"
-	"github.com/gravitational/teleport/lib/services"
-	"github.com/gravitational/teleport/lib/services/local"
-	"github.com/gravitational/teleport/lib/sshutils"
-
 	log "github.com/Sirupsen/logrus"
 	"github.com/gravitational/trace"
+
 	"golang.org/x/crypto/ssh"
-	"golang.org/x/crypto/ssh/agent"
 )
 
-type LocalKeyAgent struct {
-	agent.Agent
-	// KeyDir is the directory path where local keys are stored
+const (
+	defaultKeyDir    = ".tsh"
+	sessionKeyDir    = "sessions"
+	fileNameCert     = "cert"
+	fileNameKey      = "key"
+	fileNamePub      = "pub"
+	fileNameTTL      = ".ttl"
+	fileNameKnownCAs = "known_cas"
+)
+
+// FSLocalKeyStore implements LocalKeyStore interface using the filesystem
+// Here's the file layout for the FS store:
+// ~/.tsh/
+// ├── known_cas     --> trusted certificate authorities (their keys) in a format similar to known_hosts
```

Comment:
I suggest to rename this to `known_hosts` actually - to be fully compatible with OpenSSH and put trusted keys there. Here's an example on how we do this in tctl:

https://github.com/gravitational/teleport/blob/master/tool/tctl/main.go#L456

this will make this file 100% compatible with OpenSSH